src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2709,6 +2716,17 @@ FixedwingPositionControl::tecs_update_pitch_throttle(const float control_interva
 		     hgt_rate_sp);
 
 	tecs_status_publish(alt_sp, airspeed_sp, airspeed_rate_estimate, throttle_trim_adjusted);
+
+	// Estimate if we are in level flight.
+	// Tecs must be running, the height rate setpoint is not set or set to 0 and we are close to the target altitude and target altitude is not moving
+	if (_tecs_is_running && (!PX4_ISFINITE(hgt_rate_sp) || fabsf(hgt_rate_sp) < 0.1f)
+	    && fabsf(_current_altitude - alt_sp) < _param_nav_fw_alt_rad.get()
```

Comment:
I think these thresholds are not ideal. 
E.g. they wouldn't classify the following flight as "in level flight", as the vehicle was oscillating around the altitude setpoint due to wind/thermal conditions. 
![image](https://github.com/PX4/PX4-Autopilot/assets/26798987/f102309e-6207-4c2c-b3b7-84ded70a635d)

Could we instead look at tecs_status.altitude_reference? 
![image](https://github.com/PX4/PX4-Autopilot/assets/26798987/fb946f72-86a9-45c5-91a7-f457bb513b14)

Reply:
It would be classified as level flight! alt_sp is constant, and i guess hgt_rate_sp is NaN (Be aware that i make the estimation in the fixedWingPositionControl and hgt_rate_sp is not tecs_status/height_rate_setpoint). The hgt_rate_sp is a feedforward term i can set to TECS (which in most cases except landing/starting is NaN i think) while height_rate_setpoint is the reference for the altitude rate controller.

Reply:
ah right, my bad :+1: